src/main/java/com/configcat/Utils.java

```diff
@@ -38,6 +38,14 @@ public static DecimalFormat getDecimalFormat() {
 
     public static Config deserializeConfig(String json) {
         Config config = Utils.gson.fromJson(json, Config.class);
+
+        if (config == null ||
+                config.getEntries() == null ||
+                config.getPreferences() == null ||
+                config.getSegments() == null) {
```

Comment:
This goes against the principles we used for implementing Config V2, i.e. as little validation on deserialization as possible and how to treat missing or null map/collection properties - which are generally treated as empty.

See for example line 51 or `RolloutEvaluator.evaluateSetting`, etc.

I'd like keep this behavior for consistency and less friction for flag overrides.

```suggestion
        if (config == null) {
```